targets/frontend/src/lib/gitlab.api.js

```diff
@@ -0,0 +1,42 @@
+import subHours from "date-fns/subHours";
+
+import { request } from "./request";
+
+const url = process.env.GITLAB_URL;
+const projectId = process.env.GITLAB_PROJECT_ID;
+const accessToken = process.env.GITLAB_ACCESS_TOKEN;
+const token = process.env.GITLAB_TRIGGER_TOKEN;
+
+export function getPipelines({ ref = "master", since }) {
+  if (!since) {
+    since = subHours(new Date(), 2);
+  }
+
+  return request(
+    `${url}/projects/${projectId}/pipelines?private_token=${accessToken}&updated_after=${since.toISOString()}&ref=${ref}&order_by=updated_at`
+  );
+}
+
+export function getPipelineInfos(id) {
+  return request(
+    `${url}/projects/${projectId}/pipelines/${id}?private_token=${accessToken}`
+  );
+}
+
+export function getPipelineVariables(id) {
+  return request(
+    `${url}/projects/${projectId}/pipelines/${id}/variables?private_token=${accessToken}`
```

Comment:
i think we'd better use a header in these request so the private token dont leak in logs on the way

` --header "PRIVATE-TOKEN: <your_access_token>"`

Reply:
in fact we could already validate request using jwt

Reply:
i think we also need to use a header here (not in the url if possible)